modules/core/keeper/msg_server.go

```diff
@@ -69,17 +69,13 @@ func (k Keeper) UpdateClient(goCtx context.Context, msg *clienttypes.MsgUpdateCl
 func (k Keeper) UpgradeClient(goCtx context.Context, msg *clienttypes.MsgUpgradeClient) (*clienttypes.MsgUpgradeClientResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	upgradedClient, err := clienttypes.UnpackClientState(msg.ClientState)
-	if err != nil {
-		return nil, err
-	}
-	upgradedConsState, err := clienttypes.UnpackConsensusState(msg.ConsensusState)
-	if err != nil {
-		return nil, err
-	}
-
-	if err = k.ClientKeeper.UpgradeClient(ctx, msg.ClientId, upgradedClient, upgradedConsState,
-		msg.ProofUpgradeClient, msg.ProofUpgradeConsensusState); err != nil {
+	if err := k.ClientKeeper.UpgradeClient(
+		ctx, msg.ClientId,
+		msg.ClientState.Value,
+		msg.ConsensusState.Value,
```

Comment:
@colin-axner @damiannolan I think just passing doing `.Value` here and passing the bytes of the proton-encoded client and consensus states might not be enough. See test failing [here](https://github.com/cosmos/ibc-go/actions/runs/7849754768/job/21423773040?pr=5827), it looks like if we just pass the bytes we miss the type URL information that we need to unmarshal the bytes.

Reply:
See https://github.com/cosmos/ibc-go/pull/5827/commits/3e311639ef002c2272ed02e11cbd5aee1bc543d3

We should unmarshal to concrete types and not interfaces within the light client module.

Reply:
Thanks, Damian. I changed this to unmarshal to interface because [the tests are generally using `client types.MarshalClientState`](https://github.com/cosmos/ibc-go/blob/055a3e39c3da309e013ec4763f78d33b74be59c9/modules/core/02-client/keeper/client_test.go#L481), which [calls `cdc.MarshalInterface`](https://github.com/cosmos/ibc-go/blob/055a3e39c3da309e013ec4763f78d33b74be59c9/modules/core/02-client/types/encoding.go#L35), so when trying to unmarshal to the concrete type I would get [errors like the ones you see now in the PR](https://github.com/cosmos/ibc-go/actions/runs/7952186466/job/21706474023?pr=5827).

So I guess my question would be: do we expect the client state (and consensus state) received in the messages be marshalled as an interface or a concrete type? I am not sure if this question really makes sense, though. Maybe relayers will just send the bytes for the marshalled concrete type (it seems like [that's what hermes does](https://github.com/informalsystems/hermes/blob/f986dc82cff5aafef11befee44b1d2eb8d179ad9/crates/relayer/src/client_state.rs#L149-L152), although the [go relayer seems to marshal the interface type](https://github.com/cosmos/relayer/blob/11b7de318b5527b0f097475df1865fe472f76ff1/relayer/chains/cosmos/tx.go#L708-L716)?), so maybe we just need to change our tests to also marshal to the concrete type? Sorry, my protobuf foo is pretty weak, so I might just not making any sense, hehe. But having a bit of clarity on this would be helpful for me to unblock this work.